lib/models/device.coffee

```diff
@@ -1117,13 +1117,16 @@ exports.hasDeviceUrl = (uuid, callback) ->
 # });
 ###
 exports.getDeviceUrl = (uuid, callback) ->
-	exports.hasDeviceUrl(uuid).then (hasDeviceUrl) ->
+	exports.hasDeviceUrl(uuid).bind({}).then (hasDeviceUrl) ->
 		if not hasDeviceUrl
 			throw new Error("Device is not web accessible: #{uuid}")
 
 		return configModel.getAll().get('deviceUrlsBase')
 	.then (deviceUrlsBase) ->
-		return "https://#{uuid}.#{deviceUrlsBase}"
+		this.deviceUrlsBase = deviceUrlsBase
```

Comment:
Can you do
```coffee
.then (deviceUrlsBase) ->
	exports.get(uuid).get('uuid')
	.then (uuid) ->
		return "https://#{uuid}.#{deviceUrlsBase}"
```
instead, to avoid the magic with binding and `this` (which is actually a pretty nifty hack btw, but the simpler solution is preferable for readability :P )